src/Common/tests/System/Net/Http/Http2LoopbackConnection.cs

```diff
@@ -185,19 +186,28 @@ private async Task<Frame> ReadFrameAsync(CancellationToken cancellationToken)
             Stream oldStream = _connectionStream;
             _connectionSocket = null;
             _connectionStream = null;
-            _ignoreSettingsAck = false;
+            _ignoredSettingsAckPromise = null;
 
             return (oldSocket, oldStream);
         }
 
-        public void ExpectSettingsAck()
+        public async Task ExpectSettingsAckAsync(TimeSpan timeout)
         {
             // The timing of when we receive the settings ack is not guaranteed.
             // To simplify frame processing, just record that we are expecting one,
             // and then filter it out in ReadFrameAsync above.
 
-            Assert.False(_ignoreSettingsAck);
-            _ignoreSettingsAck = true;
+            // In case of a pending settings ack, wait for it or time out
+            var currentTask = _ignoredSettingsAckPromise?.Task;
```

Comment:
Style: This is an improper use of `var` (only allowed with `new`, hard cast, and as-cast as our common interpretation of "obvious": https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md rule 10)